### PR TITLE
Update all-keys-one-secret.md

### PR DESCRIPTION
**Pull request by MohitBishesh**
Signed-off-by: Mohit Bishesh mohitbishesh7@gmail.com

## Problem Statement

How a user using an external secrets operator can pass all the secrets or only a few as a key-value pair in the environment variables of his pod despite of that the key-values are stored either in the form of key-value or in the form of plain text over the remote secret like (GCP or AWS secrets manager or of Azure).

## Related Issue

Fixes #...
https://stackoverflow.com/questions/72487297/how-to-add-vault-secrets-to-kubernetes-env-vars 
## Proposed Changes

I had updated the readme.md file and explained

dataFrom:
  extract:
    key:
in more details

To pass a particular/all key-value secrets from the secrets manager as environment variables by adding an envFrom block in the spec of our container in the deployment file as below:

```yaml
spec:
  containers:
  - command:
    - mkdir ABC
  envFrom:
    - secretRef:
        name: 
```
How do you like to solve the issue and why?

I really want to solve this problem as recently I was working on a same use case of external secrets operator but I didn't find any proper explanation over the internet. I read different blogs, websites and even did some hit and trial and got the proper solution. I do not want someone else to face the same issues as it may stuck them and and kill their time
I feels, that after this update it will be very much useful for one using external secrets operator and understand it more easily as I have provided a detailed explanation for the same in a same page, so they will not even worry to look into multiple different pages.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by MohitBishesh** [comment]
I have signed of my commit with the commit message as 
Signed-off-by: Mohit Bishesh mohitbishesh7@gmail.com

I have identified a problem with how a user using an external secrets operator can pass all the secrets or only a few as a key-value pair in the environment variables of his pod despite that the key-values are stored either in the form of key-value or in the form of plain text over the remote secret like (GCP or AWS secrets manager or of Azure).

Also as a part of my university project, many of my colleagues are struggling in this issue and are stuck.

Please have a look at my PR so that it will be beneficial for other folks.

**Comment by Skarlso** [comment]
@MohitBishesh Hey Mohit. Only add a single `Signed-off` clause to the commit message. The email you are using to push the commit. :) That should be okay. :)

File: docs/guides/all-keys-one-secret.md [hunk 1]
```diff
@@ -18,10 +18,60 @@ Now, when creating our ExternalSecret resource, instead of using the data field,
 ```yaml
 {% include 'gcpsm-data-from-external-secret.yaml' %}
 ```
+<br>
```
Comment (knelasevero) [inline docs/guides/all-keys-one-secret.md:21]: Why use <br> here?

File: docs/guides/all-keys-one-secret.md [hunk 2]
```diff
@@ -18,10 +18,60 @@ Now, when creating our ExternalSecret resource, instead of using the data field,
...
+when we use 
 
-To check both values we can run:
+```
+  dataFrom:
+  - extract:
+      key: all-keys-example-secret
+```
+We get all the secrets present in the key-value pair in the secret manager(external secret store of GCP or AWS or Azure) in the form of key values and also we can pass either all or a few key-values as environment variables.
```
Comment (knelasevero) [inline docs/guides/all-keys-one-secret.md:31]: I don't think this description makes it clearer. 

"All secrets"? Are you implying each key value inside the secret of the external provider is a secret? That's very confusing

File: docs/guides/all-keys-one-secret.md [hunk 3]
```diff
@@ -18,10 +18,60 @@ Now, when creating our ExternalSecret resource, instead of using the data field,
...
+<br>
```
Comment (knelasevero) [inline docs/guides/all-keys-one-secret.md:32]: Please refrain from using html <br> tags here and stick to plai markdown where possible. Unless you need explicit spacings, which I don't think you need here

File: docs/guides/all-keys-one-secret.md (hunk 1 above)
Comment (MohitBishesh) [reply docs/guides/all-keys-one-secret.md]: removed the <br> tag

File: docs/guides/all-keys-one-secret.md (hunk 2 above)
Comment (MohitBishesh) [reply docs/guides/all-keys-one-secret.md]: Added a more proper explanation.

File: docs/guides/all-keys-one-secret.md (hunk 3 above)
Comment (MohitBishesh) [reply docs/guides/all-keys-one-secret.md]: I have removed all the <br> tags and used the standard markdown

File: docs/guides/all-keys-one-secret.md [hunk 4]
```diff
@@ -18,10 +18,60 @@ Now, when creating our ExternalSecret resource, instead of using the data field,
 ```yaml
 {% include 'gcpsm-data-from-external-secret.yaml' %}
 ```
+<br>
+Here, "example" is the name of Kubernetes secrets which will be created.
```
[the remark this answers is not in the document]
Comment (MohitBishesh) [reply docs/guides/all-keys-one-secret.md]: @knelasevero 
Thanks for the correction. I have done the changes and updated the PR

**Comment by MohitBishesh** [comment]
Hey @Skarlso and @knelasevero, I have updated my PR as advised by you and even clarified the description of for using key-value pairs with ESO in more simpler way. Please have a look upon so that it can be benefited by other folks and saves their time and efforts.

**Comment by knelasevero** [comment]
Hey, I'm not available until next week. Will have a look then.

In the meantime please rebase with the main branch as the PR is not mergeble and you need to sign all the commits with the same email as your GitHub email...

**Comment by MohitBishesh** [comment]
Hey @knelasevero  I have rebased the branch with the main and updated the PR. Requesting @Skarlso to please have a look!

**Comment by Skarlso** [comment]
Sorry but signing is still not right. 😔

**Comment by Skarlso** [comment]
All of the commits have to singed correctly not just the last. Either squash your commits or amend the previous commits.

**Comment by knelasevero** [comment]
Can you squash all commits into a single commit? It will make your life easier with regards to signing it, and it usually looks better on release notes

**Comment by MohitBishesh** [comment]
I sincerely apologize for my mistake, I am trying it for the first time and am learning.
I have squashed all the commits, please provide your valuable time over it.

**Comment by knelasevero** [comment]
Now that commit is changing 120 files. You probably picked commits outside of your changes to squash.

I can help you next week. Are you on kubernetes official slack?

**Comment by MohitBishesh** [comment]
Actually yes, when  I ran git rebase -i HEAD~3, I got an another commit at the top by an another author and I squashed all into that commit.
Can you guide me what I can do now?
No I am not in the official slack, can you share the invite to me.

**Comment by MohitBishesh** [comment]
Hey @knelasevero and @Skarlso now this PR is having a merge conflict with the main, I think creating a new PR will be fine now as there were some issues in Squash as well and will be easy for you to merge as well.
Is that will be okay?

**Pull request closed without merging**